src/scripts/ordo_report_mapping_annotations/create_sparql__ordo_add_annotation_based_mappings.py

```diff
@@ -0,0 +1,135 @@
+) -> Dict[str, str]:
+    """Query ontology using SPARQL query file
+
+    :param: run_command (bool): If True, then in addition to instantiating the Jinja template, it will also run the
+    instantiated queries.
+    """
+    results_dirname = os.path.basename(onto_path).replace('/', '-').replace('.', '-') + \
+        '__' + os.path.basename(query_template_path).replace('.', '-')
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    command_save_filename = 'command.sh'
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    if run_command:
+        os.makedirs(results_dirpath, exist_ok=True)
+
+    # Instantiate template
+    i = 0
+    instantiated_paths: List[str] = []
+    for mapping_precision_string, relationship_curie in mapping_str__curie__map.items():
+        i += 1
+        with open(query_template_path, 'r') as f:
+            template_str = f.read()
+        template_obj = Template(template_str)
+        instantiated_str = template_obj.render(
+            mapping_precision_string=mapping_precision_string,
+            relationship_curie=relationship_curie)
+        instantiated_query_filename = os.path.basename(query_template_path).replace('.ru.jinja2', f'{str(i)}.ru')
+        instantiated_query_path = os.path.join(SPARQL_DIR, instantiated_query_filename)
+        if run_command:
+            instantiated_query_cache_path = os.path.join(results_dirpath, instantiated_query_filename)
+            # if not (os.path.exists(instantiated_query_path) and use_cache):
+            # Save to cache
+            with open(instantiated_query_cache_path, 'w') as f:
+                f.write(instantiated_str)
+        # Save to sparql dir
+        with open(instantiated_query_path, 'w') as f:
+            f.write(instantiated_str)
+        instantiated_paths.append(instantiated_query_path)
+
+    report = {}
+    if run_command:
+        update_args = '--update ' + ' --update '.join(instantiated_paths)  # skips first --update, so add to beginning
+        command_str = f'robot query --input {onto_path} {update_args} --output {output_path}'
+
+        # Cache and run
+        # if not (os.path.exists(results_path) and use_cache):
+        with open(command_save_path, 'w') as f:
+            f.write(command_str)
+        subprocess.run(command_str.split(), capture_output=True, text=True)
+        result = subprocess.run(['ls', '-l'], capture_output=True, text=True)
+
+        # Warnings go to 'errors' as well
+        report = {'info': result.stdout, 'errors': result.stderr}
+
+    return report
+
+
+def run(input_paths: Dict[str, str], output_path: str, run_command=False):
+    """Run"""
+    # Check prerequisites
+    onto_path: str = input_paths['ontology']
+    if not os.path.exists(onto_path):
+        raise FileNotFoundError(REQS_ERR.format(onto_path))
+
+    # Get current mappings
+    report: Dict = get_report()
```

Comment:
This script, which instantiates several SPARQL update queries from a single Jinja template, depends on running another Python script which collects the latest mappings. But I'm not sure how I feel about whether or not I should have this dependency, or this is the best way to do it.